There is a fatal error on darwin that doesn't look great
```
       >   GEN    src/libexpr/primops/derivation.nix.gen.hh
       >   GEN    src/libexpr/fetchurl.nix.gen.hh
       >   CXX    src/libexpr/eval.o
       > src/libexpr/eval.cc:519:18: fatal error: 'flake/call-flake.nix.gen.hh' file not found
       >         #include "flake/call-flake.nix.gen.hh"
       >                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > 1 error generated.
       > make: *** [mk/patterns.mk:3: src/libexpr/eval.o] Error 1
       > rm src/nix/doc/files/profiles.md
       For full logs, run 'nix log /nix/store/cwhq42qdbwp2wxyqanf7q79dqsvmzpp0-nix-2.19.1.drv'.
```

--- a/pkgs/tools/package-management/nix/default.nix
+++ b/pkgs/tools/package-management/nix/default.nix
@@ -238,11 +238,8 @@ in lib.makeExtensible (self: ({
   };
 
   nix_2_19 = common {
-    version = "2.19.1";
-    hash = "sha256-OzAeQwlAF4l0h2uBWGIPvGBYNL6MpBfrdRKwHTRQXl4=";
-    patches = [
-      patch-rapidcheck-shared
-    ];
+    version = "2.19.2";
+    hash = "sha256-iA8DqS+W2fWTfR+nNJSvMHqQ+4NpYMRT3b+2zS6JTvE=";
   };
 
   # The minimum Nix version supported by Nixpkgs

--- a/pkgs/tools/package-management/nix/default.nix
+++ b/pkgs/tools/package-management/nix/default.nix
@@ -4,7 +4,6 @@
 , boehmgc
 , callPackage
 , fetchFromGitHub
-, fetchurl
 , fetchpatch
 , fetchpatch2
 , runCommand
@@ -233,6 +232,14 @@ in lib.makeExtensible (self: ({
   nix_2_18 = common {
     version = "2.18.1";
     hash = "sha256-WNmifcTsN9aG1ONkv+l2BC4sHZZxtNKy0keqBHXXQ7w=";
+    patches = [
+      patch-rapidcheck-shared
+    ];
+  };
+
+  nix_2_19 = common {
+    version = "2.19.1";
+    hash = "sha256-OzAeQwlAF4l0h2uBWGIPvGBYNL6MpBfrdRKwHTRQXl4=";
     patches = [
       patch-rapidcheck-shared
     ];

--- a/pkgs/tools/package-management/nix/default.nix
+++ b/pkgs/tools/package-management/nix/default.nix
@@ -264,7 +264,7 @@ in lib.makeExtensible (self: ({
 
   stable = addFallbackPathsCheck self.nix_2_18;
 
-  unstable = self.nix_2_18;
+  unstable = self.nix_2_19;
 } // lib.optionalAttrs config.allowAliases {
   nix_2_4 = throw "nixVersions.nix_2_4 has been removed";
 

